tests/posix/common/src/clock.c

```diff
@@ -123,3 +123,163 @@ ZTEST(posix_apis, test_realtime)
 	zassert_true(rts.tv_nsec >= tv.tv_usec * NSEC_PER_USEC,
 			"gettimeofday didn't provide correct result");
 }
+
+ZTEST(posix_apis, test_clock_nanosleep_errors_errno) {
```

Comment:
@TomFinet - thanks for your contribution!

It might be better to place this test into `tests/posix/common/src/nanosleep.c` and with that I think a lot of test code can likely be re-used with slight modification.

E.g. fector out code to an `errors_common()` function, modify the `common()` function as well. You should be able to specify a clock source and flags to the common functions.